### DISCUSSION TITLE
Possible GSoC 2003 projects.

### DISCUSSION THREAD

**Discussion by rssh** [rssh/dotty-cps-async#68] [General]
This year the scala-center participating in the Google Summer of Code program: students and open-source newcomers can receive a stipend for working on worthwhile open-source projects: https://summerofcode.withgoogle.com/

Here is a list of ideas from scala-center:    https://github.com/scalacenter/GoogleSummerOfCode/blob/main/README.md

  We have two projects in dotty-cps-async there, for which I will be a mentor.  I can hold an introduction lecture/QA session at the beginning of Marth, where I can tell you about the project and directions for possible  GSoC tasks.  If you are interested - please, register here: https://forms.gle/EQXJB5wxc8wSq4dw7,

**Comment by samadpls** [comment]
Hello @rssh 
I am interested in working on the sbt plugin development project as part of the Google Summer of Code program. I have some experience with Scala and functional programming, and I think this project would be a great opportunity for me to learn more about compiler plugins and testing frameworks.
I have registered for this session, and I am looking forward to learning more about the project and how I can contribute.

Thank you for considering my application. I am excited about the opportunity to work on this project and to be a part of the GSoC community.
Best😊,
Abdul Samad

**Comment by marrious11** [comment]
Hello sir @rssh happy new month. I hope you are fine? Our prayers for peace and support will reach Ukraine for sure.<br>
Apologies for coming in late as a student and **open-source newcomer**. Only heard of open source this year starting from January, which I started my very first open source contribution with the Ocaml community, I am interested in metaprogramming and functional programming itself which is why I took Ocaml as a starting point.<br>
I would love to express my interest to work and learn on projects from scalacenter. I stated earlier, I have basic knowledge of **OCaml and functional programming**, only hear about **scala** recently as I was exploring more into functional programming.
Furthermore, I would like to work on a project related to **#36**. I believe this project will help me familiarize with **compiler internals**, "learn more about **compiler plugins** and **testing frameworks** as well".
I could not register for this session early enough, Hopefully, I am looking forward to learning more about the project and how I can contribute as I spent more time with the community.
Not only that, but I am excited to have had the opportunity to work on the **dotty-cps-async project** and hopefully to be a part of the **scalacenter** participant in the Google Summer of Code 2023 program.

**Reply by samadpls** [reply]
To meet the minimum requirement mentioned in the issue, you need to provide an example using async and Mondas

**Reply by marrious11** [reply]
Hi @samadpls, nice to e-meet you, my friend! Would you mind taking me through some of the basic requirements, I am feeling overwhelmed feeling that I am too late for the application itself, as yesterday, is my first time to have come across GSOC, where I found Scala itself too.

**Reply by marrious11** [reply]
One more thing @samadpls : could you help me list out the two projects that are listed here https://forms.gle/EQXJB5wxc8wSq4dw7, I find it hard to open the link, perhaps it could be deprecated

**Reply by samadpls** [reply]
It's completely normal to feel overwhelmed. To get started with this project, you need to learn the basics of Scala first. Additionally, you should also learn about monads, how they work, and what they are used for. It may be true that it's a bit late now, but you can still try learning Scala or explore other programming language projects.

I don't mean to discourage you, but it may be too late to look for a project at this point. Many mentors have already chosen whom to select for their GSoC project, and even if they haven't, understanding the flow of the project in such a short amount of time can be challenging, especially if you are new to the language. Today is already April 2nd, and applications will close on April 4th. It may not be possible for most of us to understand the flow and write a complete proposal mentioning the tasks needed to complete this project in just two days.

Perhaps you can use this time as an opportunity to learn Scala and apply for GSoC next year. I was also interested in this project, but I haven't received any updates from Rulsan after our first discussion.  best of luck for your future endeavors

**Reply by marrious11** [reply]
Thank you! At least you have been of help today, let me use this time left as an opportunity to learn scala like you rightly said.
 I have some knowledge about functional programming with Ocaml already, I guess the syntax would not be too different, right!

**Reply by samadpls** [reply]
Your Welcome and yeah they kinda look same, for you it will be much easy to work with scala,

**Reply by marrious11** [reply]
:smiley_cat: Yeah, let me give my best short, though. 
Concerning the example you sated as a basic requirement @samadpls, It is found on one of the issues? If `YES`, could you reference it for me ! 
The issues sir @rssh referenced for me earlier today were not understood because I couldn't trace them back on the repo.

**Reply by samadpls** [reply]
this one #6

**Reply by marrious11** [reply]
Thank you :smiling_face_with_tear:

**Reply by rssh** [reply]
> I was also interested in this project, but I haven't received any updates from Rulsan after our first discussion. 
  
 The last reply was in https://github.com/rssh/dotty-cps-async/issues/6#issuecomment-1462556112, I.e.  I think that I have not received updates ;)
   
My apologies about the non-clear 'qualification task':  Next year, it will be better if we will participate in the next GSoC.

  I recommend choosing some themes [programming language,  project,  etc] and becoming familiar with them now, during the year.  Then near each big project time from time provides some connected opportunities that can be discovered.
